src/gtimelog/main.py

```diff
@@ -1035,46 +984,15 @@ def on_send_report(self, action, parameter):
 
     def send_email(self, sender, recipient, subject, body):
         smtp_server = self.gsettings.get_string('smtp-server')
+        smtp_port = self.gsettings.get_int('smtp-port')
         smtp_username = self.gsettings.get_string('smtp-username')
-        callback = functools.partial(self._send_email, sender, recipient, subject, body)
+        mail_protocol = self.gsettings.get_string('mail-protocol')
+        callback = functools.partial(send_email, mail_protocol, smtp_server, smtp_port, sender, recipient, subject, body)
         if smtp_username:
             start_smtp_password_lookup(smtp_server, smtp_username, callback)
         else:
             callback('')
 
-    def _send_email(self, sender, recipient, subject, body, smtp_password):
```

Comment:
I need to pause a bit and think about this.

There was one function `send_email()` once.  Then it had to do an asynchronous password lookup, so it was split into two halves.  The two halves were purposefully kept together.  And now one half suddenly becomes an API.

Reply:
Ok, so I thought briefly, and I think I want to keep `_send_email` here, but have it be a mostly empty wrapper that looks things up in self.gsettings and calls `send_email()`, and also passes all the necessary parameters as keyword arguments.